Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -13,17 +13,87 @@
 
 #include <cstdint>
 #include <vector>
+#include <array>
+#include <map>
 
 namespace o2
 {
 namespace trd
 {
 
-class DigitIndex;
+class Digit;
 
-typedef std::uint16_t Digit;
-typedef std::vector<Digit> DigitContainer;
-typedef std::vector<DigitIndex> DigitIndexContainer;
+constexpr int kTB = 30;
+constexpr int KEY_MIN = 0;
+constexpr int KEY_MAX = 2211727;
+
+typedef std::uint16_t ADC_t;                              // the ADC value type
+typedef std::array<ADC_t, kTB> ArrayADC_t;                // the array ADC
+typedef std::vector<Digit> DigitContainer_t;              // the digit container type
+typedef std::map<int, ArrayADC_t> ArrayADCMapContainer_t; // a map container type for signal handling during digitization
+
+class Digit
+{
+ public:
+  Digit() = default;
+  ~Digit() = default;
+  Digit(const int det, const int row, const int pad, const ArrayADC_t adc)
+    : mDetector(det), mRow(row), mPad(pad), mADC(adc) {}
+  // Copy
+  Digit(const Digit&) = default;
+  // Assignment
+  Digit& operator=(const Digit&) = default;
+  // Modifiers
+  void setDetector(int det) { mDetector = det; }
+  void setRow(int row) { mRow = row; }
+  void setPad(int pad) { mPad = pad; }
+  void setADC(ArrayADC_t adc) { mADC = adc; }
+  // Get methods
+  int getDetector() const { return mDetector; }
+  int getRow() const { return mRow; }
+  int getPad() const { return mPad; }
+  ArrayADC_t getADC() const { return mADC; }
+
+  // Set of static helper methods
+  static int calculateKey(const int det, const int row, const int col) { return ((det << 12) | (row << 8) | col); }
+  static int getDetectorFromKey(const int key) { return (key >> 12) & 0xFFF; }
+  static int getRowFromKey(const int key) { return (key >> 8) & 0xF; }
+  static int getColFromKey(const int key) { return key & 0xFF; }
+  static void convertMapToVectors(ArrayADCMapContainer_t& adcMapCont,
+                                  DigitContainer_t& digitCont)
+  {
+    //
```

Comment:
Please use `const&` for constant input here and I believe there is no need having these function inline (or implemented right here at the declaration).

Reply:
Thanks for spotting the missing `const&`, I will update the definition. For the inline declaration, this is all that should be here by now; should I move this to `Digit.cxx`?.

Reply:
I added the missing `const&`, should I move the function inlines to the declaration?